atuin-client/src/database.rs

```diff
@@ -326,6 +396,31 @@ mod test {
     use super::*;
     use std::time::{Duration, Instant};
 
+    macro_rules! assert_search_eq {
+        ($db:expr, $mode:expr, $query:expr, $expected:expr) => {
+            let results = $db.search(None, $mode, $query).await.unwrap();
+            assert_eq!(
+                results.len(),
+                $expected,
+                "query \"{}\", commands: {:?}",
+                $query,
+                results.iter().map(|a| &a.command).collect::<Vec<&String>>()
+            );
+        };
+        ($db:expr, $mode:expr, $query:expr, $expected:expr, $commands:expr) => {
+            let results = $db.search(None, $mode, $query).await.unwrap();
+            let commands: Vec<&String> = results.iter().map(|a| &a.command).collect();
+            assert_eq!(
+                results.len(),
+                $expected,
+                "query \"{}\", commands: {:?}",
+                $query,
+                commands
+            );
+            assert_eq!(commands, $commands);
+        };
+    }
+
```

Comment:
Does this need to be a macro? or can it be a regular async function?

Reply:
Sure, I've updated the code to use `async func` but the requirement to `.await.unwrap()` all calls makes the assertions longer again. Maybe a `try_join!( assertions )` could be used instead?

Reply:
I see, that's fine IMO